gitbook/configuration/forwarding.md

```diff
@@ -1,41 +1,40 @@
 ## Agent Forwarding
-When using `ssh` to connect to a remote server there might be the use case
-where the user or an application on the remote server wants to receive an access token from
+When using `ssh` to connect to a remote server, you may also need to receive access tokens from
 the local agent. This is possible by forwarding the UNIX domain socket used for
-communicating with the agent. This can be done using the `-R` option of
-`ssh`. Example:
+communicating with the agent.
+
+This needs a **client** and **server** side configuration. Both
+configurations can be done by a local user.
+
+### Client
+This can be done using the `-R` option of
+`ssh` (e.g. with `ssh -R /tmp/oidc-forward:$OIDC_SOCK  user@host`).
+Defining an alias makes this more easily usable:
+. Example:
 ```
-ssh -R /tmp/oidc-forward:$OIDC_SOCK  user@host
+alias ssh='ssh -R /tmp/oidc-forward-$RANDOM:$OIDC_SOCK'
+ssh user@host
 ```
-However, if you do this you still have to manually export the `OIDC_SOCK`
-environment variable on the server (`export
-OIDC_SOCK=/tmp/oidc-forward`).
+For most convenience, you should put the alias line into your `.profile`, or `.zshrc`.
+
 
-**To use agent forwarding we recommend the following configurations:**
-Put the following in your `.bash_profile` on the server:
+### Server
+On the server, you have to set the `OIDC_SOCK`
+environment variable (`export OIDC_SOCK=/tmp/oidc-forward`).
+
+**We recommend the following configurations:**
+Put the following in your `.profile`, `.zshrc`, or `.bash_profile` on the server:
 ```
 test -z $OIDC_SOCK && {
     export OIDC_SOCK=`/bin/ls -rt /tmp/oidc-forward-* 2>/dev/null | tail -n 1`
 }
+alias ssh='ssh -R /tmp/oidc-forward-$RANDOM:$OIDC_SOCK'
 ```
-and this into your `.bash_logout` on the server:
+
+Add this into your `.zlogout`, or `.bash_logout` on the server:
 ```
 if [ -e $OIDC_SOCK ]; then
     rm -f $OIDC_SOCK
 fi
 ```
 
-Add this to `.bash_aliases` on your local machine:
-```
-alias OA='echo -R /tmp/oidc-forward-$RANDOM:$OIDC_SOCK'
-alias ssh-oidc="ssh \`OA\`"
```

Comment:
I liked the approach with the `ssh-oidc` alias. Do you really think we should guide users to overwriting their `ssh` command and use agent forwarding all the time?

Reply:
Well... I think users that understand nothing will benefit from it working.

Users that understand the concept will choose a different alias.

I'm about to change mine to overwrite ssh

Reply:
I'm still against suggesting this is a default for inexperienced people. People that want to do it can simply decide themselves that they want to overwrite ssh, as you said they are experienced enough to adapt the instructions.
But for others it should be easy to and straightforward to do `ssh` for ssh and `ssh-oidc` for ssh with oidc forwarding.

There are side-effects and also maybe security-implications when doing the ssh with oidc-agent forwarding, at very least every ssh connection creates a new entry in the server's `/tmp/`:
```
srw-------  1 gabriel  gabriel     0 Feb 14 10:23 oidc-forward-9443=
srw-------  1 gabriel  gabriel     0 Feb 14 10:24 oidc-forward-5998=
```

Reply:
Ok

Reply:
Not super trivial, since you need to be root, though

Reply:
In general yes, but for the special case where a unix user is shared by multiple human users this is entirely possible without being root.
(That situation might not be a recommended setup, but it certainly is used)

Reply:
Unfortunately yes.